src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,68 @@
+@import "../utils/mixins";
+@import "../utils/variables";
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: 706px;
+
+  &__day {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+
+    @include square;
+
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+
+    background-color: #eee;
+    border: 1px solid black;
+
+    cursor: pointer;
+    transition: background-color 0.5s;
+  }
+
+  &__day:hover {
+    transform: translateY(-20px);
+    background-color: pink;
+  }
+
+  @for $var from 1 to 32 {
+    &__day--#{$var}::before {
+      content: "#{$var}";
+    }
+  }
+
+  &--month-length--28 {
+    @for $var from 29 through 31 {
+      .calendar__day--#{$var} {
+        display: none;
+      }
+    }
+  }
+
+  &--month-length--29 {
+    @for $var from 30 through 31 {
+      .calendar__day--#{$var} {
+        display: none;
+      }
+    }
+  }
+
+  &--month-length--30 {
+    .calendar__day--31 {
+      display: none;
+    }
+
+  }
```

Comment:
What is the reason for these loops? All these rules can be written in one @for loop.

Reply:
This is a correct tip, thank you, I forgot about this.